### DISCUSSION TITLE
[ZK Stack] - List of currently deployed smart contracts (Diamond Facets)

### DISCUSSION THREAD

**Discussion by sekuba** [zkSync-Community-Hub/zksync-developers#323] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [X] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.75.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [X] Windows
- [ ] Other (Please specify below)

### Issue Description

Is there a list of current Facets used by the DiamondProxy on mainnet? I am trying to map them out but so far was unable to find especially the GettersFacet and AdminFacet. I figured i could use the loupe of the GettersFacet to go deeper afterwards?

### Expected Behavior

There should be a list of deployed smart contracts somewhere.

### Current Behavior

No such list, can only find some contracts via google/etherscan.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by edevaldonunez** [comment]
To obtain a list of current Facets used by the DiamondProxy on the mainnet, you can utilize tools like Etherscan or directly query the blockchain using Ethereum JSON-RPC calls. The deployed contracts, including the DiamondProxy and its facets, should be visible on the blockchain explorer. Additionally, you can inspect the contract source code or interact with the contracts programmatically to determine their functionality and facets.

**Reply by sekuba** [reply]
wow this sounds like an AI answer. Etherscan does not list facets when you only have their diamonds address. see below for the addresses for anyone else wondering

**Comment by sekuba** [comment, marked as answer]
I went deeper and found out some things on the way in case anyone has similar noob questions ;)
As you cannot easily derive the diamond facets on etherscan, there is nice tools that can, for example [louper.dev](https://louper.dev). (This site also lets you interact with the facets functions like with contracts on etherscan)

In case you are using some cmdline tool for interacting with the diamond you can use the diamond's address but the ABI of the facet whose function you're trying to call. Usually diamonds have a loupe facet (in zksync era gettersFacet) that has a `facetAddresses()` function to list all current facets

### Current important addresses:

Matter Labs MultiSig:
https://etherscan.io/address/0x4e4943346848c4867f81dfb37c4ca9c5715a7828

Main diamond
https://etherscan.io/address/0x32400084C286CF3E17e7B677ea9583e60a000324

Current Facets:
ExecutorFacet https://etherscan.io/address/0x3a4ef67c6cab51444e5d3861843f7f4a37f64f0a (freezable)
AdminFacet https://etherscan.io/address/0xe6426c725cb507168369c10284390e59d91ec821 (not freezable)
GettersFacet https://etherscan.io/address/0xc4a5e861df9dd9495f8dba1c260913d1a9b8ec2b (not freezable)
MailboxFacet https://etherscan.io/address/0x0f58fd6c9ed966e09c1dffbc8e6ff600ec65f6eb (freezable)

Verifier:
https://etherscan.io/address/0x3390051435eCB25a9610A1cF17d1BA0a228A0560

Governance (for optional timelock and transparency):
https://etherscan.io/address/0x0b622A2061EaccAE1c664eBC3E868b8438e03F61

Validator Timelock:
https://etherscan.io/address/0xa0425d71cb1d6fb80e65a5361a04096e0672de03

Validator:
https://etherscan.io/address/0x3527439923a63f8c13cf72b8fe80a77f6e572092

L1ERC20Bridge:
https://etherscan.io/address/0x810c6598caaa08b61f6430df5a8e120b3390d78a
  proxy: https://etherscan.io/address/0x57891966931eb4bb6fb81430e6ce0a03aabde063

**Comment by jodychvs** [comment]
To access the list of current Facets used by the DiamondProxy on mainnet, you can typically find this information by examining the project's source code or documentation. If not available, reaching out to the project maintainers or community forums for assistance might provide the necessary insights. #DiamondProxy #SmartContracts
